power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -1003,6 +1109,51 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
         return se_input;
     }
 
+    template <bool sym>
+    std::vector<ShortCircuitInput> prepare_short_circuit_input() {
+        assert(is_topology_up_to_date_ && is_parameter_up_to_date<sym>());
+
+        std::vector<IdxVector> topo_fault_indices(state_.math_topology.size());
+        std::vector<IdxVector> topo_bus_indices(state_.math_topology.size());
+
+        for (Idx fault_idx{0}; fault_idx < state_.components.template size<Fault>(); ++fault_idx) {
+            auto const& fault = state_.components.template get_item_by_seq<Fault>(fault_idx);
```

Comment:
we only consider the faults which has `status()` returns `true`.  So we need a check here. If the `status` is `false`, we `continue`.